### PR TITLE
Sync with main SDK for project upgrades feature

### PR DESCRIPTION
**Pull request by stwiname**
# Description
Sync with main SDK to get new project upgrades feature and POI rework

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have tested locally
- [x] I have performed a self review of my changes
- [x] Updated any relevant documentation
- [x] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] My code is up to date with the base branch
- [ ] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

File: packages/common-near/src/project/versioned/v1_0_0/model.ts [hunk 1]
```diff
@@ -80,25 +71,19 @@ export class ProjectNetworkDeploymentV1_0_0 {
   @Transform(({value}: TransformFnParams) => value.trim())
   @IsString()
   chainId: string;
-  @ValidateNested()
-  @Type(() => FileType)
-  @IsOptional()
-  chaintypes?: FileType;
   @IsOptional()
   @IsArray()
   bypassBlocks?: (number | string)[];
```
Comment (jiqiang90) [inline packages/common-near/src/project/versioned/v1_0_0/model.ts]: bypassBlocks?: (number | `${number}-${number}`)[];

File: packages/node/src/main.ts [hunk 2]
```diff
@@ -16,16 +16,10 @@ initLogger(
...
-const { bootstrap } = require('./init');
-if (
-  !(
-    argv._[0] === 'test' ||
-    argv._[0] === 'mmr-migrate' ||
-    argv._[0] === 'mmr-regen' ||
-    argv._[0] === 'force-clean'
-  )
-) {
+// Check for no subcommand
```
Comment (jiqiang90) [inline packages/node/src/main.ts:19]: do we need this too
```
(BigInt.prototype as any).toJSON = function () {
  return this.toString();
};
```

File: packages/node/src/indexer/dictionary.service.ts [hunk 3]
```diff
@@ -21,18 +22,18 @@ export type SpecVersionDictionary = {
 };
 
 @Injectable()
-export class DictionaryService
-  extends CoreDictionaryService
-  implements OnApplicationShutdown
-{
+export class DictionaryService extends CoreDictionaryService {
```
Comment (jiqiang90) [inline packages/node/src/indexer/dictionary.service.ts:13]: I suppose line 13-19, specVersions is not used in Near?

File: packages/node/src/indexer/indexer.manager.ts [hunk 4]
```diff
@@ -173,8 +166,8 @@ export class IndexerManager extends BaseIndexerManager<
 
   private async indexReceipt(
     receipt: NearTransactionReceipt,
-    dataSources: SubqlProjectDs[],
-    getVM: (d: SubqlProjectDs) => Promise<IndexerSandbox>,
+    dataSources: NearProjectDs[],
+    getVM: (d: NearProjectDs) => Promise<IndexerSandbox>,
```
Comment (jiqiang90) [inline packages/node/src/indexer/indexer.manager.ts:170]: line 184 `baseCustomHandlerFilter` not used?

**Review by jiqiang90 — CHANGES_REQUESTED** [review]

File: packages/node/src/main.ts (hunk 2 above)
Comment (stwiname) [reply packages/node/src/main.ts:19]: I don't think so. Near doesn't seem to use much BigInt

File: packages/node/src/indexer/indexer.manager.ts (hunk 4 above)
Comment (stwiname) [reply packages/node/src/indexer/indexer.manager.ts:170]: Its used by `BaseIndexerManager`

**Review by jiqiang90 — APPROVED** [review]

File: packages/types/package.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
     "/dist"
   ],
   "dependencies": {
+    "@subql/types-core": "^0.1.0",
```
Comment (jiqiang90) [inline packages/types/package.json]: It is 0.1.1 now

**Review by jiqiang90 — CHANGES_REQUESTED** [review]

**Review by jiqiang90 — APPROVED** [review]

**Pull request merged**